### PR TITLE
test against a FIPS config

### PR DESCRIPTION
**Pull request by reaperhulk**
Let's see what happens.

### CODE REVIEW

**Comment by reaperhulk** [comment]
This isn't actually testing against FIPS yet because I forgot to pass the env var to docker. And it won't work because `tox` uses MD5 internally to track package hashes, which causes the entire thing to explode.

`tox` also uses sha256 so it's unclear if they need md5 though. Maybe it's just legacy and can be removed?

**Comment by simo5** [comment]
> This won't work because `tox` uses MD5 internally to track package hashes, which causes it to fail while trying to build the env to run the tests.
> 
> https://github.com/tox-dev/tox/blob/a8b34cc5690cef34fa91ca0f1950a1fdde0f8b72/src/tox/logs/env.py#L34-L42

Solved here: https://github.com/tox-dev/tox/pull/1385

**Comment by reaperhulk** [comment]
(Once tox does a release I can pick this back up easily)

**Comment by reaperhulk** [comment]
Looks like for whatever reason this isn't actually activating the FIPS mode. We pass the env var to docker and it's passed through to the toxenv so it should be working, but clearly I've missed something.

**Comment by reaperhulk** [comment]
This can be rebased and we can check coverage once https://github.com/pyca/infra/pull/244 merges.

File: src/cryptography/hazmat/backends/openssl/backend.py [hunk 1]
```diff
@@ -122,14 +123,24 @@ def __init__(self):
...
+            self.activate_osrandom_engine()
         self._dh_types = [self._lib.EVP_PKEY_DH]
         if self._lib.Cryptography_HAS_EVP_PKEY_DHX:
             self._dh_types.append(self._lib.EVP_PKEY_DHX)
 
     def openssl_assert(self, ok):
         return binding._openssl_assert(self._lib, ok)
 
+    def _fips_enabled(self):
+        fips_mode = getattr(self._lib, "FIPS_mode", lambda: 0)
```
[the remark this answers is not in the document]
Comment (reaperhulk) [reply src/cryptography/hazmat/backends/openssl/backend.py]: So either way we need to clear the error stack after we call, okay

**Pull request closed without merging**